ivadomed/transforms.py

```diff
@@ -148,6 +148,9 @@ def __call__(self, sample, metadata, data_type='im'):
             return None, None
         else:
             for tr in self.transform[data_type].transforms:
+                if isinstance(tr, NormalizeInstance):
+                    numpy_to_tensor = NumpyToTensor()
+                    sample, metadata = numpy_to_tensor(sample, metadata)
```

Comment:
Thanks @cakester. I've changed my mind: instead of before the `NomarlizeInstance`, could you please force the `NumpyToTensor` as last transform?
Reason: The current solution would not work if, for some reason, the user do not use `NormalizeInstance` --> we still need our samples to be moved from Numpy to Tensor.

Reply:
I tried moving NumpyToTensor last, it was failing a bunch of tests then I tried
 ```            numpy_to_tensor = NumpyToTensor()
            numpy_to_tensor_called = False
            for tr in self.transform[data_type].transforms:
                if isinstance(tr, NormalizeInstance):
                    sample, metadata = numpy_to_tensor(sample, metadata)
                    numpy_to_tensor_called = True
                sample, metadata = tr(sample, metadata)

            if not numpy_to_tensor_called:
                sample, metadata = numpy_to_tensor(sample, metadata)
            return sample, metadata```
and looked at the test results, the test_automated_training fails nevertheless (same failure message), mind taking a look as I do not know how to resolve that error? thanks

Reply:
Can you please copy the error in this thread? Cheers

But my guess is: Have you removed the `NumpyToTensor` from the config file used by the testing? If not --> that's the reason: this transform is called twice: you get an error on the second call because the sample is already a Tensor.

The solution you implemened looks good. But you now need to remove `NumpyToTensor` from all the config files, documentation and testing. Cheers

Reply:
I tried removing NumpyToTensor from automate_training_config.json and I got this 
```
--- Logging error ---
Traceback (most recent call last):
  File "/Users/runner/work/ivadomed/ivadomed/ivadomed/scripts/automate_training.py", line 95, in train_worker
    ivado.run_command(config, thr_increment=thr_incr)
  File "/Users/runner/work/ivadomed/ivadomed/ivadomed/main.py", line 407, in run_command
    debugging=context["debugging"])
  File "/Users/runner/work/ivadomed/ivadomed/ivadomed/training.py", line 158, in train
    for i, batch in enumerate(train_loader):
  File "/Users/runner/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/torch/utils/data/dataloader.py", line 345, in __next__
    data = self._next_data()
  File "/Users/runner/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/torch/utils/data/dataloader.py", line 385, in _next_data
    data = self._dataset_fetcher.fetch(index)  # may raise StopIteration
  File "/Users/runner/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/torch/utils/data/_utils/fetch.py", line 44, in fetch
    data = [self.dataset[idx] for idx in possibly_batched_index]
  File "/Users/runner/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/torch/utils/data/_utils/fetch.py", line 44, in <listcomp>
    data = [self.dataset[idx] for idx in possibly_batched_index]
  File "/Users/runner/work/ivadomed/ivadomed/ivadomed/loader/loader.py", line 754, in __getitem__
    data_type="im")
  File "/Users/runner/work/ivadomed/ivadomed/ivadomed/transforms.py", line 151, in __call__
    sample, metadata = tr(sample, metadata)
  File "/Users/runner/work/ivadomed/ivadomed/ivadomed/transforms.py", line 50, in wrapper
    return wrapped(self, sample, metadata)
  File "/Users/runner/work/ivadomed/ivadomed/ivadomed/transforms.py", line 76, in wrapper
    return wrapped(self, sample, metadata)
  File "/Users/runner/work/ivadomed/ivadomed/ivadomed/transforms.py", line 889, in __call__
    metadata["elastic"] = [None, None]
TypeError: list indices must be integers or slices, not str```

Reply:
> I tried removing NumpyToTensor from automate_training_config.json and I got this

Which commit? Can't see. Thanks

Reply:
I ran locally and this was what I got, I'll upload it again

Reply:
Yes okay I see.
So yep that's a good starting point for your investigations. Good luck with this, really appreciated. Let us know how you go. Cheers

Reply:
do you know how to fix the issue with automate_training test with the error above?

Reply:
@cakester If I may suggest, running the test within the debugger might give you clues as to where the problem is. Moving forward, running the debugging will also give you more insights about the functional/integration aspects of ivadomed (ie: "big picture" of the project)